satellite/gracefulexit/pathcollector.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gracefulexit
+
+import (
+	"context"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/metainfo"
+)
+
+var _ metainfo.Observer = (*PathCollector)(nil)
+
+// PathCollector uses the metainfo loop to add paths to node reservoirs
+//
+// architecture: Observer
+type PathCollector struct {
+	db        DB
+	nodeIDMap map[storj.NodeID]struct{}
```

Comment:
nit, this could maybe just be nodeIDs instead of nodeIDMap

Reply:
i'm a ex-java person.  we are wordy :) .  i'll make a note to revisit in another PR